typescript/lib/config.ts

```diff
@@ -208,8 +208,11 @@ export class AIConfigRuntime implements AIConfig {
    * @param saveOptions Options that determine how to save the AIConfig to the file.
    */
   public save(filePath?: string, saveOptions?: SaveOptions) {
+    const keysToOmit = ['filePath'] as const;
```

Comment:
Hmm, do we want the serialized AIConfig to have dynamic top-level fields that we want to persist or will it always match the keys defined in the interface (excluding the open `[k: string]: any;`)?

If we want it to match the interface instead of the runtime interface,  rather than omitting specific fields here, it may be easier to just copy the exact fields we want persisted from the serialized schema. That way the runtime can have any number of added fields without having to worry about coming here and omitting them.

Something like:
```
const aiConfigObj: AIConfig = {
    name: this.name;
    description: this.description,
    schema_version: this.schema_version;
    metadata: {
      ...this.metadata, // could technically cloneDeep this but doesn't look like this function is modifying anything anyway so doesn't seem necessary
    },
    prompts: [
      ...this.prompts.map(prompt => {
        if (saveOptions.serializeOutputs === true) {
          return prompt;
        } else {
        return {
          ...prompt,
          outputs: [],
         }
       }
     })
    ],
}
```

Otherwise, if we want to support persisting dynamic/generic values on the config other than filePath and callbackManager then this should be ok to start. Just a couple suggestions for improvements in that case:

1. We might also want to support an `omitKeys` array in SaveOptions for users to exclude keys they've defined on the config as well. Probably not necessary for v0, but might be worth adding an Issue for.
2. We could have an `_internal` key on the config type which contains `filePath`, `callbackManager`, etc. for the internal implementation that we know we don't want to persist. Then this can be done simply by omitting `_internal` without tracking any keys to omit here

Reply:
Adding to Linear Task